internal/injector/builtin/yaml/go-main.yml

```diff
@@ -21,7 +21,41 @@ aspects:
       - prepend-statements:
           imports:
             tracer: gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer
+            profiler: gopkg.in/DataDog/dd-trace-go.v1/profiler
+            os: os
+            slices: slices
+            strings: strings
           template: |-
             tracer.Start(tracer.WithOrchestrion(map[string]string{"version": {{printf "%q" Version}}}))
             defer tracer.Stop()
 
+            // TODO: this probably premature, not sure if "autoinjection"
+            // will actually be a thing for Go binaries.
+            injected := slices.Contains(
```

Comment:
side note while I pass through this PR: you must enforce a separate lexical scope scope for those statements to avoid name clashes with the rest of the main function we don't know about:

```
{
    // your instrumentation block
}
```

@RomainMuller often uses the [IIFE](https://en.wikipedia.org/wiki/Immediately_invoked_function_expression) pattern for this too, but the block should be enough from what I can see here.

Reply:
Good to know, thanks! Looks like it's already put in its own scope: https://github.com/DataDog/orchestrion/pull/178/files#diff-c4820fd0f6a6380e158a2a9cd02a46aa7da2df4879c4f9914ab76075d9f5498aR24-R49